src/simulation/elements/HSWC.cpp

```diff
@@ -70,6 +74,14 @@ int Element_HSWC::update(UPDATE_FUNC_ARGS)
 						else if (parts[r>>8].life==0)
 							parts[r>>8].life = 10;
 					}
+					if (parts[i].tmp == 1 && (r & 0xFF) == PT_FILT || (r & 0xFF) == PT_PHOT || (r & 0xFF) == PT_BRAY && (r & 0xFF) != PT_HSWC)
```

Comment:
This logic isn't doing what you think it is doing, you need some parenthesis here. Also, the != PT_HSWC at the end is useless (once you fix the logic)

Also, can you change all r & 0xFF instances to r&0xFF and all r >> 8 to r>>8? Usually don't have a space there because of how often we do it